src/Common/src/System/Text/DBCSDecoder.cs

```diff
@@ -103,7 +104,7 @@ public unsafe override int GetCharCount(byte* bytes, int count, bool flush)
             if (count < 0)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
 
-            bool excludeLastByte = count > 0 || (!flush && IsLastByteALeadByte(bytes, count));
+            bool excludeLastByte = count > 0 && !flush && IsLastByteALeadByte(bytes, count);
```

Comment:
Is there a test we can add to exercise this? (I'm assuming it's not being exercised as the logic here is changing.)